### PR TITLE
Implement Background2d

### LINKED ISSUE #47: Implement Background rate IRF (symmetric case)
Description here: https://open-gamma-ray-astro.github.io/gamma-astro-data-formats/irfs/full_enclosure/bkg/index.html

### PR DESCRIPTION
**Pull request by MaxNoe**
fixes #47 

This is for now the radially symmetric version in the field of view

The tests for if gammpy supports reading is marked as xfail due to the axis order inconsistencies in gammapy, see 

https://github.com/gammapy/gammapy/issues/3025
https://github.com/gammapy/gammapy/issues/1221
https://github.com/gammapy/gammapy/issues/2067

### CODE REVIEW

File: pyirf/io/gadf.py [hunk 1]
```diff
@@ -212,10 +212,73 @@ def create_energy_dispersion_hdu(
...
+)
+def create_background_2d_hdu(
+    background_2d,
+    reco_energy_bins,
+    fov_offset_bins,
+    extname="BACKGROUND",
+    **header_cards,
+):
+    """
+    Create a fits binary table HDU in GADF format for the PSF table.
```
Comment (HealthyPear) [inline pyirf/io/gadf.py:235]: [...] for the **background** table.

File: pyirf/io/gadf.py [hunk 2]
```diff
@@ -212,10 +212,73 @@ def create_energy_dispersion_hdu(
...
+    See the specification at
+    https://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/full_enclosure/psf/psf_table/index.html
```
Comment (HealthyPear) [inline pyirf/io/gadf.py:237]: https://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/full_enclosure/bkg/index.html#bkg-2d

File: pyirf/io/gadf.py [hunk 3]
```diff
@@ -212,10 +212,73 @@ def create_energy_dispersion_hdu(
...
+            "THETA_HI": u.Quantity(fov_offset_bins[1:], ndmin=2).to(u.deg),
+            # transpose as FITS uses opposite dimension order
+            "BKG": background_2d.T[np.newaxis, ...].to(GADF_BACKGROUND_UNIT),
+        }
+    )
+
+    # required header keywords
+    header = DEFAULT_HEADER.copy()
+    header["HDUCLAS1"] = "RESPONSE"
+    header["HDUCLAS2"] = "PSF"
```
Comment (HealthyPear) [inline pyirf/io/gadf.py:269]: BKG

**Review by HealthyPear — CHANGES_REQUESTED** [review]

File: pyirf/io/tests/test_gadf.py [hunk 4]
```diff
@@ -85,3 +84,33 @@ def test_psf_table():
...
+    for point_like in [True, False]:
+        with tempfile.NamedTemporaryFile(suffix='.fits') as f:
+            hdu = create_background_2d_hdu(background, e_bins, fov_bins)
+
+            fits.HDUList([fits.PrimaryHDU(), hdu]).writeto(f.name)
+
+            # test reading with gammapy works
+            bg2d = Background2D.read(f.name, 'BACKGROUND')
+
+            assert u.allclose(background, bg2d.data.data, atol=1e-16 * u.Unit('TeV-1 s-1 sr-1'))
```
Comment (adonath) [inline pyirf/io/tests/test_gadf.py:116]: I think for the `Background2D` class in Gammapy the axis order is equivalent, so you can actually implement the test, no?

```python3
from astropy import units as u
from gammapy.irf import Background2D

energy = [0.1, 10, 1000] * u.TeV
offset = [0, 1, 2, 3] * u.deg
data = np.zeros((2, 3)) * u.Unit("s-1 MeV-1 sr-1")
data.value[1, 0] = 2
data.value[1, 1] = 4

bkg = Background2D(
    energy_lo=energy[:-1],
    energy_hi=energy[1:],
    offset_lo=offset[:-1],
    offset_hi=offset[1:],
    data=data,
)

print(bkg.data.data.shape)
print([ax.name for ax in bkg.data.axes])
```
Which prints:
```
(2, 3)
['energy', 'offset']
```

Comment (MaxNoe) [reply pyirf/io/tests/test_gadf.py:116]: The test here is meant to verify that the file / hdu can be read. So creating the in-memory version does not really help here.

**Comment by MaxNoe** [comment]
Ok, the copy paste leftovers are fixed and I added a test for the background rate calculation.

File: pyirf/io/tests/test_gadf.py (hunk 4 above)
Comment (MaxNoe) [reply pyirf/io/tests/test_gadf.py:116]: We opted to in-memory have the numpy convention for axis order like in the GADF doc page, which means we transpose when reading and writing from / to fits.

**Comment by MaxNoe** [comment]
@HealthyPear could you review again? I think this is done.

**Review by HealthyPear — APPROVED** [review]
Just a reminder for future testing: wouldn't be easier to define a common toy event data set for all irf unit-tests? or 2, one for point-like and one for full-enclosure

**Comment by MaxNoe** [comment]
@HealthyPear maybe, but here I chose the values so that the results can be easily checked by hand, I don't know if it is possible to come up with a single event set for all IRF tests with that property.

**Pull request merged**